app_grim_reefer/src/buzzer.c

```diff
@@ -0,0 +1,65 @@
+#include "buzzer.h"
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(buzzer);
+
+K_MSGQ_DEFINE(buzzer_msgq, sizeof(enum buzzer_cond), 10, 1);
+
+void buzzer_tell(enum buzzer_cond cond) { k_msgq_put(&buzzer_msgq, &cond, K_MSEC(10)); }
+
+#define BUZZER_THREAD_STACK_SIZE 500
+#define BUZZER_THREAD_PRIORITY   10
+
+// 2 second beepcode, 2 second pause
+const uint16_t beepcodes[] = {
+    [buzzer_cond_noflash] = 0b1010101010101010,
+    [buzzer_cond_low_battery] = 0b1110001110001110,
+    [buzzer_cond_missing_sensors] = 0b1100000011000000,
+    [buzzer_cond_landed] = 0b1111111100000000,
+    [buzzer_cond_ok] = 0b0000000000000000,
+};
```

Comment:
Why not just make this an enum in buzzer.h?

Reply:
The buzzer enum has a relatively low number of elements so it'll get backed by an 8 bit value. The beep codes are just 16 bit long bit strings and would overflow the 8 bit value and end up with half their data truncated. 

https://godbolt.org/z/v1evfza44

Also when you're editing the tones you don't have to wait for a full recompile bc of a header change lol.

Reply:
Sounds good. Btw might be recommended to compile for arm gcc and not x86 since this code is for a different target

Reply:
Ooh this is cool.